src/BcPrebidVast.js

```diff
@@ -0,0 +1,365 @@
+	    				var cpm = 0.0;
+	    				$$PREBID_GLOBAL$$.prebid_creativ = null;
+	    				for (var i = 0; i < arrBids.length; i++) {
+	    					if (arrBids[i].cpm > cpm) {
+	    						cpm = arrBids[i].cpm;
+	    						$$PREBID_GLOBAL$$.prebid_creative = arrBids[i].vastUrl;
+	    					}
+	    				}
+	    	            BC_prebid_in_progress = false;
+						dispatchPrebidDoneEvent();
+	    			}
+    			}
+    			else {
+    				// no bids
+    	            BC_prebid_in_progress = false;
+					dispatchPrebidDoneEvent();
+    			}
+    			_logger.log(_prefix, 'Selected VAST url: ' + $$PREBID_GLOBAL$$.prebid_creative);
+			});
+    	}
+    };
+    pbjsScr.onerror = function(e) {
+    	_logger.error(_prefix, 'Failed to load prebid.js. Error event: ', e);
+		if (options.pageNotificationCallback) {
+			options.pageNotificationCallback('message', 'Failed to load prebid.js');
+		}
+    	$$PREBID_GLOBAL$$.bc_pbjs_error = true;
+		dispatchPrebidDoneEvent();
+    };
+    pbjsScr.async = true;
+    pbjsScr.type = 'text/javascript';
+    pbjsScr.src = options.prebidPath ? options.prebidPath : '//acdn.adnxs.com/prebid/not-for-prod/1/prebid.js';
+    var node = document.getElementsByTagName('head')[0];
+    node.appendChild(pbjsScr);
+}
+
+// this function loads MailOnline Plugin
+var molLoadingInProgress = false;
+var molLoaded = false;
+function loadMolPlugin(callback) {
+	var vjs = window.videojs || false;
+	if (!vjs) {
+    	_logger.warn(_prefix, 'Videojs is not loaded yet');
+		callback(false);
+		return;
+	}
+	if (!vjs.getPlugins().vastClient) {
+		if (document.getElementById('mol-script')) {
+			if (!molLoadingInProgress) {
+		    	_logger.log(_prefix, 'MailOnline Plugin ' + (molLoaded ? '' : 'not ') + 'loaded successfilly already');
+				callback(molLoaded);
+			}
+			else {
+				var waitMolLoaded = setInterval(function() {
+					if (!molLoadingInProgress) {
+						clearInterval(waitMolLoaded);
+				    	_logger.log(_prefix, 'MailOnline Plugin ' + (molLoaded ? '' : 'not ') + 'loaded successfilly already');
+						callback(molLoaded);
+					}
+				}, 50);
+			}
+			return;
+		}
+		molLoadingInProgress = true;
+	    var molScr = document.createElement('script');
+	    molScr.id = 'mol-script';
+	    molScr.onload = function() {
+	    	_logger.log(_prefix, 'MailOnline Plugin loaded successfilly');
+	    	molLoaded = true;
+	    	molLoadingInProgress = false;
+	    	callback(true);
+	    };
+	    molScr.onerror = function(e) {
+	    	_logger.error(_prefix, 'Failed to load MailOnline Plugin. Error event: ', e);
+	    	molLoadingInProgress = false;
+	    	callback(false);
+	    };
+	    molScr.async = true;
+	    molScr.type = 'text/javascript';
+	    molScr.src = '//acdn.adnxs.com/video/plugins/mol/videojs_5.vast.vpaid.min.js';
+	    var node = document.getElementsByTagName('head')[0];
+	    node.appendChild(molScr);
+	}
+	else {
+    	_logger.log(_prefix, 'MailOnline Plugin already loaded');
+		callback(true);
+	}
+}
+
+(function () {
```

Comment:
why is this in an self-Invoking Anonymous Function?

Reply:
For immediate execution (loading prebid.js ASAP), only one time execution, not to keep in the scope after execution. It is not a big win, and, if you think it will confuse someone, I can rewrite this code to not use anonymous function.